provider/ec2/environ.go

```diff
@@ -425,18 +422,21 @@ func (e *environ) StartInstance(args environs.StartInstanceParams) (_ *environs.
 	// attached to the machine must all match, and must be the same
 	// as specified zone (if any).
 	availabilityZone, placementSubnetID, err := e.deriveAvailabilityZoneAndSubnetID(args)
-	switch {
-	case err != nil && errors.IsNotValid(err):
-		return nil, errors.Wrap(err, environs.ErrAvailabilityZoneFailed)
-	case err != nil:
-		return nil, err
+	if err != nil {
+		return nil, &common.StartInstanceError{
+			Err:             errors.Trace(err),
+			ZoneIndependent: !errors.IsNotValid(err),
+		}
 	}
 
 	arches := args.Tools.Arches()
 
 	instanceTypes, err := e.supportedInstanceTypes()
 	if err != nil {
-		return nil, errors.Trace(err)
+		return nil, &common.StartInstanceError{
+			Err:             errors.Trace(err),
+			ZoneIndependent: true,
```

Comment:
Is the supported types a regional thing? (So while a given zone may not have an instance of a type, it still knows the type)

Reply:
Availability is dependent on region, but there's also capacity for AZs I'm pretty sure. We don't check that up front (not sure if you can?) - so there's no AZ specific error coming from that method call.

Reply:
Don't these need to use the ZoneIndependentError function?